arc_cache.go

```diff
@@ -42,11 +45,27 @@ func newARCCachedBS(ctx context.Context, bs Blockstore, lruSize int) (*arccache,
 	return c, nil
 }
 
+func (b *arccache) getLock(k cid.Cid) *sync.RWMutex {
+	return &b.lks[mutexKey(k)]
+}
+
+func mutexKey(k cid.Cid) uint8 {
+	return k.KeyString()[len(k.KeyString())-1]
+}
+
 func (b *arccache) DeleteBlock(k cid.Cid) error {
+	if !k.Defined() {
+		return ErrNotFound
```

Comment:
This should return nil (or a different error). `DeleteBlock` is idempotent and never returns "not found".